### ISSUE TITLE
Create Next.js project & deploy on Firebase

### ISSUE THREAD

**Issue by longwind48** [bettersg/SchemesSG_v3#32]
## TL;DR:
- You can start developing on Next.js without worrying about deployment. See repo https://github.com/bettersg/SchemesSG_v3/tree/stg/frontend
- Any changes pushed to stg will auto-deploy to the Firebase staging app: https://schemessg-v3-dev.web.app/
- Production deployment is coming soon.

I've added a vanilla Next.js 14 w/ app router to our repo in stg branch, and set up GitHub Actions to automatically deploy to Firebase Hosting for our staging environment. This is part of my task to deploy both our frontend and backend to the new Firebase space. Right now, any commits made to the stg branch will auto-deploy to the Firebase staging app at https://schemessg-v3-dev.web.app/. Production deployment will be set up soon.

## New workflow for frontend devs:
- Create a feature branch from stg.
- Make your changes and test them locally (npm run dev).
- To test and build for staging, run (npm run test-build:staging)
- Push your changes to your branch and create a PR to stg.
- Once approved and merged, your changes will auto-deploy to the staging app.
- For production: The production app will be set up soon. Once that’s ready, creating a PR from stg to main will trigger the deployment to production.

I chose Next.js for its strong support for different rendering strategies, which is great for SEO and speed. However, if anyone has strong opinions about moving to Vite or another framework, we can discuss and accommodate as needed.
